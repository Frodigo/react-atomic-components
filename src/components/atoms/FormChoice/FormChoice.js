import React from 'react';
import PropTypes from 'prop-types';

import FormTitle from '../FormTitle';

import classes from './FormChoice.module.css';

/**
 * This is the FormTitle.
 * Component that display form label/legend
 * @param {*} props
 */
const FormChoice = (props) => {
    const { type, isChecked, id, name, label, ...rest } = props;

    return (
        <div className={classes.formChoiceField}>
            <input
                type={type}
                name={name}
                id={id}
                checked={isChecked}
                className={`${classes.choice} ${classes[type]}`}
                {...rest}
            />
            <FormTitle htmlFor={id}>{label}</FormTitle>
        </div>
    );
};

FormChoice.propTypes = {
    id: PropTypes.string.isRequired,
    isChecked: PropTypes.bool,
    label: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    type: PropTypes.oneOf(['checkbox', 'radio'])
};

FormChoice.defaultProps = {
    type: 'checkbox'
};

export default FormChoice;
