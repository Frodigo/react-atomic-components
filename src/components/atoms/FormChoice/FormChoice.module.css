.formChoiceField {
    position: relative;
    overflow: hidden;
}

.choice {
    left: -999px;
    position: absolute;
}

.choice + label {
    align-items: center;
    cursor: pointer;
    display: inline-flex;
    justify-content: flex-start;
}

.choice + label::before {
    border: 1px solid var(--color-secondary);
    content: '';
    display: inline-flex;
    height: 20px;
    margin-right: 10px;
    width: 20px;
}

.choice:checked + label::after {
    background-color: var(--color-secondary);;
    content: '';
    display: block;
    height: 16px;
    left: 3px;
    position: absolute;
    top: 6px;
    width: 16px;
}


.radio {
    composes: choice;
    
}

.radio + label::before {
    border-radius: 50%;
}

.radio:checked + label::after {
    border-radius: 50%;
}
