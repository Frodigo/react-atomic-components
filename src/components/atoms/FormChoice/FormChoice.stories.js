import React from 'react';

import { default as FormChoice } from './FormChoice';

export default {
    title: 'Atoms/Form/FormChoice',
    component: FormChoice,
    argTypes: {}
};

const Template = (args) => <FormChoice {...args} />;

export const Checkbox = Template.bind({});
Checkbox.args = {
    id: 'inputId',
    label: 'This is the checkbox'
};

export const Radio = Template.bind({});
Radio.args = {
    id: 'inputId',
    type: 'radio',
    name: 'radio1',
    label: 'This is the radio button'
};
