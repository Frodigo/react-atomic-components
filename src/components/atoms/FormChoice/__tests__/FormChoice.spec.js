import React from 'react';
import { render, getByLabelText } from '@testing-library/react';

import FormChoice from '../FormChoice';

test('Checkbox renders correctly', () => {
    const props = {
        id: 'inputId',
        label: 'label',
        name: 'input'
    };

    const { container } = render(<FormChoice {...props} />);

    const input = getByLabelText(container, 'label');

    expect(container).toMatchSnapshot();
    expect(input.getAttribute('id')).toStrictEqual('inputId');
});

test('Checkbox is disabled', () => {
    const props = {
        id: 'inputId',
        label: 'label',
        disabled: true,
        name: 'input'
    };

    const { container } = render(<FormChoice {...props} />);

    const input = getByLabelText(container, 'label');

    expect(input.getAttribute('disabled')).toBeDefined();
});

test('Checkbox is checked', () => {
    const props = {
        id: 'inputId',
        label: 'label',
        isChecked: true,
        name: 'input'
    };

    const { container } = render(<FormChoice {...props} />);

    const input = getByLabelText(container, 'label');

    expect(input.getAttribute('checked')).toBeDefined();
});

test('Radio button renders correctly', () => {
    const props = {
        id: 'inputId',
        label: 'label',
        type: 'radio',
        name: 'input'
    };

    const { container } = render(<FormChoice {...props} />);

    const input = getByLabelText(container, 'label');

    expect(container).toMatchSnapshot();
    expect(input.getAttribute('id')).toStrictEqual('inputId');
});

test('Radio button is disabled', () => {
    const props = {
        id: 'inputId',
        label: 'label',
        disabled: true,
        type: 'radio',
        name: 'input'
    };

    const { container } = render(<FormChoice {...props} />);

    const input = getByLabelText(container, 'label');

    expect(input.getAttribute('disabled')).toBeDefined();
});

test('Radio button is selected', () => {
    const props = {
        id: 'inputId',
        label: 'label',
        isChecked: true,
        type: 'radio',
        name: 'input'
    };

    const { container } = render(<FormChoice {...props} />);

    const input = getByLabelText(container, 'label');

    expect(input.getAttribute('checked')).toBeDefined();
});
