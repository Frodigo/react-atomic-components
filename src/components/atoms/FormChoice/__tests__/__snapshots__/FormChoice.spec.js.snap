// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders correctly 1`] = `
<div>
  <div
    class="formChoiceField"
  >
    <input
      class="choice checkbox"
      id="inputId"
      name="input"
      type="checkbox"
    />
    <label
      class="label"
      for="inputId"
    >
      label
    </label>
  </div>
</div>
`;

exports[`Radio button renders correctly 1`] = `
<div>
  <div
    class="formChoiceField"
  >
    <input
      class="choice radio"
      id="inputId"
      name="input"
      type="radio"
    />
    <label
      class="label"
      for="inputId"
    >
      label
    </label>
  </div>
</div>
`;
